sdl2promela/promela/generator.py

```diff
@@ -0,0 +1,182 @@
+from multipledispatch import dispatch
+from typing import List, Set, Text, Tuple, Type, TextIO
+
+from . import model
+
+INDENT = "  "
+
+class Context:
+    stream : TextIO
+    indents : List[str]
+    pending_indent : bool
+
+    def __init__(self, stream : TextIO):
+        self.stream = stream
+        self.indents = []
+        self.pending_indent = False
+
+    def push_indent(self, indent : str):
+        self.indents.append(indent)
+
+    def pop_indent(self):
+        self.indents.pop()
+
+    def _get_indent(self):
+        return "".join((self.indents))
+
+    def output(self, data : str):
+        if self.pending_indent:
+            print(self._get_indent(), file=self.stream, end='')
```

Comment:
you know me - I see duplication I extract a method ;) ( `self.indent()` )

Reply:
I'd even consider two level of extraction - `self.print` that uses the stream and "end" parameter

Reply:
ack